perf_workflow/run_perf_suite/ccr_perf_test.py

```diff
@@ -0,0 +1,294 @@
+        self.execution_id = None
+        self.test_succeeded = False
+        self.failure_reason = None
+
+    def get_infra_repo_url(self):
+        if "GITHUB_TOKEN" in os.environ:
+            return "https://${GITHUB_TOKEN}@github.com/opensearch-project/opensearch-infra.git"
+        else:
+            return "https://github.com/opensearch-project/opensearch-infra.git"
+
+    def run(self):
+        self.setup_logging()
+        if self.tests_result_dir is not None:
+            tests_dir = self.tests_result_dir + "/" + str(self.test_num)
+        else:
+            tests_dir = os.path.join(
+                str(os.getcwd()).split("cross-cluster-replication")[0]
+                + "cross-cluster-replication/build/test-results/perf-test/"
+                + str(self.test_num)
+            )
+            if os.path.exists(tests_dir):
+                shutil.rmtree(tests_dir)
+        os.makedirs(tests_dir, exist_ok=True)
+        with TemporaryDirectory(keep=False, chdir=True) as work_dir:
+            current_workspace = os.path.join(work_dir.name, "infra")
+            with GitRepository(self.get_infra_repo_url(), "main", current_workspace):
+                with WorkingDirectory(current_workspace):
+                    cluster_config = PerfTestClusterConfig(
+                        self.security,
+                        self.test_config.get("DataNodes", 1),
+                        self.test_config.get("MasterNodes", 0),
+                    )
+                    # TODO: Add support for configurable instance type. Default is m5.2xlarge
+                    with PerfTestCluster.create(
+                        self.bundle_manifest,
+                        self.config,
+                        self.test_id + "-l",
+                        cluster_config,
+                        current_workspace,
+                    ) as leaderCluster, PerfTestCluster.create(
+                        self.bundle_manifest,
+                        self.config,
+                        self.test_id + "-f",
+                        cluster_config,
+                        current_workspace,
+                    ) as followerCluster:
+                        # Wait for cluster to be up.
+                        leaderCluster.wait_for_processing(tries=5, delay=30, backoff=2)
+                        followerCluster.wait_for_processing(
+                            tries=5, delay=30, backoff=2
+                        )
+                        self.setup_replication(
+                            leaderCluster, followerCluster, self.test_config["Workload"]
+                        )
+                        # Setup args for Perf Suite.
+                        target_hosts = {
+                            "default": [leaderCluster.endpoint_with_port()],
+                            "follower": [followerCluster.endpoint_with_port()],
+                        }
+                        workload_options = self.workload_options(
```

Comment:
As discussed - lets makes sure, we have the same instance type across the runs and the workload params are configured to match the instance type.

Reply:
Yep. All tests are being run on `m5.2xlarge`